draft-pauly-dprive-oblivious-doh.md

```diff
@@ -248,14 +244,16 @@ Content-Type header in the response before processing the payload. A response wi
 treated as an error and be handled appropriately. All other aspects of the HTTP response and error handling are
 inherited from standard DoH.
 
-Proxies MUST forward any Target responses with 2xx, 3xx, 4xx, or 5xx status codes unmodified to the Client.
-Note that if a Client receives a 3xx status code and chooses to follow a redirect, the subsequent request
-MUST also be performed through a Proxy in order to avoid directly exposing requests to the Target.
-Target responses with 1xx status codes MUST NOT be forwarded to the Client.
-If a Proxy receives a successful response from a Target without the "application/oblivious-dns-message"
+Proxies forward responses from the Target to client, without any modifications to the body. In general,
+the Proxy forwards the status code received from the Target to the Client. The Proxy also SHOULD add a
+Proxy-Status response header with an "received-status" parameter indicating the status code sent by the Target.
+If a Proxy receives a successful (2xx) response from a Target without the "application/oblivious-dns-message"
 HTTP Content-Type header, it MUST return a 502 (Bad Gateway) response to the Client request, along with
```

Comment:
Flagging this, since Mark suggested that this be a 415 (Unsupported Media Type) along with http_request_error Proxy Error Type.

Reply:
Right, although I don't think that's right since it is an error in the Target response, not the Client request.

Reply:
Yep, I concur. I'm just noting it so @mnot doesn't think we ignored it.

Reply:
Ah, right - I misread. However, it's still not a `http_protocol_error`. If you want to use proxy-status here, you'll probably need to mint your own proxy error type; happy to help define it.

Reply:
I guess it's a question of if it's worth defining a new proxy-status error, which I'd probably call "http_response_content_type", or we just get rid of this entire area of text and let the bad 2xx response flow back to the client and let the client realize there's an error with the content.

@mnot thoughts?

Reply:
Letting the client deal with it is definition the lowest friction way forward. If you define a new proxy error type, it'd be good to think about the right level of specificity...

Reply:
Cool, I've updated the text to just remove this and say that the proxy forwards the status code along. Please take a look, @mnot.